### DISCUSSION TITLE
更改组件状态的优先级问题

### DISCUSSION THREAD

**Discussion by happyElina** [alibaba/formily#2016] [Q&A 2.x] [answered]
目前我们修改组件状态，可以是通过form的state来修改：
```javascript
form.setState((state) => {
      state.readPretty = !state.readPretty
})
```

也可以通过form.setFieldState 修改具体表单项的状态：
```javascript
form.setFieldState('select3', (state) => {
     state.readPretty = false;
 })
```
现在我发现调用了form.setFieldState修改了某一项的readPretty 之后，这一项的readPretty 就不再受form.setState的管控了，请问这块的逻辑有文档里说明吗？暂时没有找到

或者目前的设计就是，一旦表单项的某个属性通过form.setFieldState修改了，form.setState 就不再对这一项起作用吗？

**Comment by janryWang** [comment, marked as answer]
你可以设置pattern = null，这样会继承上下文，否则它相当于是自身管理，就跟css一样

**Reply by happyElina** [reply]
就是系统不支持form 和 field 同时控制某个字段的状态吧，应该避免这种操作？

**Reply by janryWang** [reply]
不是避免，是根据业务需求来定的，要么走继承，要么走自控，不能同时控制的，因为你不知道哪个优先级高

**Reply by happyElina** [reply]
好的
